### ISSUE TITLE
Add Missing Methods for Custom Distributions to Ensure Turing.jl Compatibility

### ISSUE THREAD

**Issue by kiante-fernandez** [itsdfish/SequentialSamplingModels.jl#17]
Hey @itsdfish. @DominiqueMakowski  and I noticed that the codebase creating custom distributions suitable for use with Turing.jl may be missing some methods. This omission can result in errors when users attempt to estimate parameters using Turing. [see this thread](https://discourse.julialang.org/t/am-i-creating-distributions-wrong-in-turing/60142/12). 

To address this issue, I suggest adding the following methods to each distribution, following the guidelines outlined in the [Distributions.jl documentation](https://juliastats.org/Distributions.jl/stable/extends/#Univariate-Distribution).

1. minimum(d::UnivariateDistribution)
2. maximum(d::UnivariateDistribution)
3. insupport(d::UnivariateDistribution, x::Real)
4. params(d::UnivariateDistribution)

While adding these methods, we can improve the usability, functionality, and standardization of using sequential sampling model distributions for use with Turing.

I am able to get started on this and opening a pull request with the necessary changes. Please let me know if there is any way I can assist or contribute further.

Best,

Kianté

**Comment by itsdfish** [comment]
Hi Kianté,

That sounds good.  I've never encountered a problem so far.  As a result, I never implemented those methods. Are you aware of any specific issues?

I think you might be able to use `method(d::SequentialSamplingModel)` for `minimum`, `maximum`, and `insupport`. The function `params` would need to be dispatched on each type. 

Another thing to be aware of is that `ContinuousUnivariateDistribution` is technically not appropriate for some of the models. Unfortunately, Distributions.jl does not have a general type that matches, for example, the LBA. What it likely needs is something like `MixedMultivariateDistribution`, which would be discrete for choice and continuous for reaction time. I discussed the design pattern of Distributions.jl with a few people on Slack, but I did not really understand how it works, and what the implications would be fore defining a new generic distribution type.
